File: server/src/models/equipment.model.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Equipment = sequelize.define('Equipment', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  serial_number: {
    type: DataTypes.STRING,
    unique: true,
  },
  enterprise_id: {
    type: DataTypes.UUID,
    allowNull: false,
  },
  status: {
    type: DataTypes.ENUM('functional', 'broken', 'lost', 'returned'),
    defaultValue: 'functional',
  },
  received_date: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
  notes: {
    type: DataTypes.TEXT,
  }
}, {
  tableName: 'equipment',
  timestamps: true,
});

module.exports = Equipment;

File: lib/features/workflow/equipment/equipment_model.dart
import 'equipment_entity.dart';

class EquipmentModel extends EquipmentEntity {
  EquipmentModel({
    required super.id,
    required super.name,
    super.serialNumber,
    required super.enterpriseId,
    super.status = EquipmentStatus.functional,
    required super.receivedDate,
    super.notes,
  });

  factory EquipmentModel.fromJson(Map<String, dynamic> json) {
    return EquipmentModel(
      id: json['id'],
      name: json['name'],
      serialNumber: json['serial_number'],
      enterpriseId: json['enterprise_id'],
      status: _parseStatus(json['status']),
      receivedDate: DateTime.parse(json['received_date']),
      notes: json['notes'],
    );
  }

  static EquipmentStatus _parseStatus(String? status) {
    switch (status) {
      case 'broken': return EquipmentStatus.broken;
      case 'lost': return EquipmentStatus.lost;
      case 'returned': return EquipmentStatus.returned;
      default: return EquipmentStatus.functional;
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'serial_number': serialNumber,
      'enterprise_id': enterpriseId,
      'status': status.name,
      'received_date': receivedDate.toIso8601String(),
      'notes': notes,
    };
  }
}
